numpy/distutils/command/build.py

```diff
@@ -27,6 +36,17 @@ def initialize_options(self):
         old_build.initialize_options(self)
         self.fcompiler = None
         self.warn_error = False
+        self.cpu_baseline = "min"
+        self.cpu_dispatch = "max -xop -fma4" # drop AMD legacy features by default
+        self.disable_optimization = False
+        """
+        the '_simd' module is a binary monster, adding more disaptched features
+        will increase binary size and compiling time so by default we minimize
+        the targeted features to the most common used by NumPy SIMD interface(NPYV),
+        NOTE: any specified features will be ignored if they're part of the baseline
+        or not supported by compiler or platform.
+        """
+        self.simd_test = "SSE2 SSE42 (FMA3 AVX2) AVX512F AVX512_SKX VSX VSX2 VSX3 NEON ASIMD"
```

Comment:
@rgommers, @mattip, another issue about the binary size is here too, the SIMD module is actually a bunch of submodules and each submodule contains python methods that provide access to universal intrinsics depend on the domain of targeted CPU feature that represented by the submodule.

The targeted features can be chosen via build option `--simd-test`, on X86_64, and assuming the compiler has the full support of them, then
the size of the shared object of SIMD module will be **1216kb** without debugging symbols.

Reply:
I will release a patch to [MacPython/numpy-wheels](https://github.com/MacPython/numpy-wheels) to disable all tested targets since the goal from this module to test any new intrinsics by the contributor through CI. @rgommers, @mattip, Do u agree with me fellas?

Reply:
That makes sense to me, thanks @seiko2plus. Without debugging symbols this won't be very helpful if there's an issue on an end user machine; if there's a need later on we could provide a test wheel outside of PyPI somewhere with `simd_test` including debug symbols.

Reply:
I did some modifications to allow pass an empty string through `--simd-test` in order to disable all targets, also I added `SSE41` and `XOP`(in case if it enabled through --cpu-dispatch) to the default dispatched targets.

why outside wheel of PyPi? we can just add them with different suffixes, right?, for example:
  - `numpy-dbg`: include debugging symbols and `simd_tests` targets.
  - `numpy-avx2f`: NumPy with AVX2 and FMA3 as a baseline `--cpu-baseline` `avx2+fma3`, that should increase the performance for non-optimized areas since baseline compiler flags pass it all sources.
  - `numpy-vsx3`: same as avx2f but for power9 users.
 -  `numpy`: the default